lib/sessions.js

```diff
@@ -0,0 +1,68 @@
+'use strict';
+
+const redis = require('redis');
+const session = require('express-session');
+const connectRedis = require('connect-redis');
+const cookieParser = require('cookie-parser');
+
+module.exports = (app, config) => {
+
+  const encryption = require('./encryption')(config.session.secret);
+
+  const RedisStore = connectRedis(session);
+  const client = redis.createClient(config.redis.port, config.redis.host);
+
+  client.on('error', (e) => {
+    throw e;
+  });
+
+  const store = new RedisStore({
+    client: client,
+    ttl: config.session.ttl,
+    secret: config.session.secret,
+    serializer: {
+      parse: (data) => JSON.parse(encryption.decrypt(data)),
+      stringify: (data) => encryption.encrypt(JSON.stringify(data))
+    }
+  });
+
+  let secure;
+
+  app.use((req, res, next) => {
+    secure = req.protocol === 'https' ? true : false;
+    next();
+  })
+
+  if (secure) {
+    app.set('trust proxy', 1);
+  }
```

Comment:
is this code executed at all? `secure` is defined only at the first request, whereas its value is used here when the file is loaded.